components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -918,7 +905,133 @@ void AdsServiceImpl::MigratePrefsVersion2To3() const {
 }
 
 int AdsServiceImpl::GetPrefsVersion() const {
-  return profile_->GetPrefs()->GetInteger(prefs::kVersion);
+  return GetIntegerPref(prefs::kVersion);
+}
+
+bool AdsServiceImpl::GetBooleanPref(
+    const std::string& path) const {
+  auto* prefs = profile_->GetPrefs();
+  auto value = prefs->GetBoolean(path);
+
+  if (prefs->HasPrefPath(path)) {
+    // Return previously persisted value
+    return value;
+  }
+
+  // Persist and return value
+  prefs->SetBoolean(path, value);
+  return value;
+}
+
+void AdsServiceImpl::SetBooleanPref(
+    const std::string& path,
+    const bool value) {
+  profile_->GetPrefs()->SetBoolean(path, value);
+}
+
+int AdsServiceImpl::GetIntegerPref(
+    const std::string& path) const {
+  auto* prefs = profile_->GetPrefs();
+  auto value = prefs->GetInteger(path);
+
+  if (prefs->HasPrefPath(path)) {
+    // Return previously persisted value
+    return value;
+  }
+
+  // Persist and return value
+  prefs->SetInteger(path, value);
+  return value;
+}
+
+void AdsServiceImpl::SetIntegerPref(
+    const std::string& path,
+    const int value) {
+  profile_->GetPrefs()->SetInteger(path, value);
+}
+
+double AdsServiceImpl::GetDoublePref(
+    const std::string& path) const {
+  auto* prefs = profile_->GetPrefs();
+  auto value = prefs->GetDouble(path);
```

Comment:
I don't really get this logic. You first get it from path and then you check the same path. If path exists you return value otherwise you set value. But if the path doesn't exist wouldn't `value` be empty (you set it and return it). I don't follow why we would need to set anything in get (get is set as `const`).

Reply:
See next comment, I will add more to the comments